### ISSUE TITLE
Index of array 'icolamp' above upper bound in FORTRAN generate_events tests

### ISSUE THREAD

**Issue by valassi** [madgraph5/madgraph4gpu#710]
See WIP MR #709 and https://github.com/madgraph5/madgraph4gpu/pull/709/files#diff-89aea48b80945c9e9b75afc0c26896c5876c54b3a2d37e22e4a8e0a11d2e8c65

```
Fortran runtime error: Index '108' of dimension 2 of array 'icolamp' above upper bound of 105
```

Could this be the same issue @oliviermattelaer was observing, discussed in #655?

**Comment by oliviermattelaer** [comment]
Yes looks like the same issue. So revert your revert and it should work

**Comment by valassi** [comment]
Hi @oliviermattelaer thanks a lot.

Indeed I included your patch and this is fixed.

This is a compromise: without your patch, we get this #710 and other errors you found for ATLAS/CMS, with your patch we get #655. I prefer to fix this, and be stuck with #655 that will need some debugging.

I am running more tests and will soon merge #709 including the move to a gpucpp including your patch. Then I will close this issue. Thanks again
Andrea
